Sources/_MatchingEngine/Regex/Parse/Parse.swift

```diff
@@ -172,34 +196,88 @@ extension Parser {
     return .concatenation(.init(result, loc(_start)))
   }
 
+  /// Perform a recursive parse for the branches of a condional.
+  mutating func parseConditionalBranches(
+    start: Source.Position, _ cond: AST.Conditional.Condition
+  ) throws -> AST {
+    let child = try parse()
```

Comment:
We could, though we'd need some extra care to handle corner cases such as an empty body, and we'd need extra logic to emit a diagnostic for too many branches

Reply:
Alternation sides can be empty (e.g. a successful empty match), so I don't know how much extra logic we'd need here.

Reply:
To be clear, I'm referring to the empty-body case e.g `(?(1))`. Is there any reason to avoid doing the recursive parse as it currently is? IMO the behavior we want falls fairly neatly out of it.